plugin/src/main/java/org/opensearch/ml/utils/MLExceptionUtils.java

```diff
@@ -59,4 +59,12 @@ public static void logException(String errorMessage, Exception e, Logger log) {
             log.error(errorMessage, e);
         }
     }
+
+    public static Throwable getRootCause(Throwable throwable) {
+        Throwable cause = throwable;
+        while (cause.getCause() != null && cause != cause.getCause()) {
+            cause = cause.getCause();
+        }
+        return cause;
+    }
```

Comment:
Recommend removing this as the handling has been implemented in #2577

Reply:
Yup, I removed it.